src/Symfony/Component/Validator/ConstraintValidator.php

```diff
@@ -87,8 +87,14 @@ protected function formatValue($value, $format = 0)
     {
         if (($format & self::PRETTY_DATE) && $value instanceof \DateTimeInterface) {
             if (class_exists('IntlDateFormatter')) {
+                $timezone = $value->getTimezone();
+                // Timezone 'Z' is a valid ISO8601 indicator for UTC
+                $timezone = 'Z' === mb_strtoupper($timezone->getName()) ? new \DateTimeZone('UTC') : $timezone;
+                // Only use timezone if it's valid, otherwise fall back to system default
+                $timezone = \in_array($timezone->getName(), \DateTimeZone::listIdentifiers(), true) ? $timezone : null;
```

Comment:
Are we sure that this can happen? If so, can you add a test for this case also that would fail otherwise?

Reply:
@xabbuh I've added a test to check that invalid timezones are handled. Due to not knowing the timezone of the system running it's only valid if the formatter is present.

Reply:
In the case where the timezone is invalid, shouldn't we trigger a deprecation that we won't support it anymore? It seems wrong to keep passing null.

Reply:
That can probably be addressed in 4.4. This PR is because the previous change has broken BC in the 4.3 branch. Prior to 4.3.5 nothing/null was passed to the formatter, it makes sense to keep this behaviour intact.

Reply:
What about caching the list in a static variable? I fear this call can be heavy in a loop.

Reply:
Also we don't need to check if it's valid in the case we set it ourselves to UTC isn't it?